lib/taiko.js

```diff
@@ -345,33 +359,53 @@ module.exports.client = () => clientProxy;
  * Allows switching between tabs using URL or page title.
  *
  * @example
- * await switchTo('https://taiko.gauge.org/') # switch using URL
- * await switchTo('Taiko') # switch using Title
+ * # switch using URL
+ * await switchTo('https://taiko.gauge.org/')
+ * # switch using Title
+ * await switchTo('Taiko')
+ * # switch using regex URL
+ * await switchTo(/http(s?):\/\/(www?).google.(com|co.in|co.uk)/)
+ * # switch using regex Title
+ * await switchTo(/Go*gle/)
  *
  * @param {string} targetUrl - URL/Page title of the tab to switch.
  *
  * @returns {Promise}
  */
+
 module.exports.switchTo = async targetUrl => {
   validate();
   let t = typeof targetUrl;
-  if (t !== 'string') {
+  let q = Object.prototype.toString
+    .call(targetUrl)
+    .includes('RegExp');
+  let isRegexText = '';
+  if (t != 'string' && !q) {
     throw new TypeError(
-      'The "targetUrl" argument must be of type string. Received type ' +
+      'The "targetUrl" argument must be of type string or regex. Received type ' +
         t,
     );
   }
+  if (q) {
+    targetUrl = new RegExp(targetUrl);
+    isRegexText = ' Regex';
+  }
   const targets = await targetHandler.getCriTargets(
```

Comment:
@zabil , Actually `/regexString instanceof RegExp/` always returns false in our case i.e `targetObject instanceof RegExp` is always false.

This is the reason i guess : https://stackoverflow.com/questions/4339288/typeof-for-regexp#comment4724685_4339350

That's the reason i had to do something like this :
https://github.com/o-x-y-g-e-n/taiko/blob/738fb2c2d47cf3ae87273f5e1abc86a9c16a9098/lib/taiko.js#L371-L374

I am not able to know the exact reason as per what is happening, so the workaround would be instead of `/regexString/ instanceof RegExp` , the condition would be `Object.prototype.toString.call(targetUrl).includes('RegExp')`

Reply:
That is weird, good catch. I understand the need for this check.

I've added a few comments for making the error/info messages common to avoid conditions on generating them. I guess once that is done this PR should be good.